diff --git a/aggregation/src/ivc/error.rs b/aggregation/src/ivc/error.rs
index e187767..334d0cb 100644
--- a/aggregation/src/ivc/error.rs
+++ b/aggregation/src/ivc/error.rs
@@ -25,6 +25,7 @@ pub enum IvcError {
 
 impl From<plonk::Error> for IvcError {
     fn from(e: plonk::Error) -> Self {
+        eprintln!("R3 IVC synthesis underlying error: {e:?}");
         IvcError::ProofGeneration(e)
     }
 }
